velox/functions/sparksql/Arithmetic.h

```diff
@@ -318,4 +318,62 @@ struct IsNanFunction {
     }
   }
 };
+
+FOLLY_ALWAYS_INLINE static int8_t fromHex(char c) {
+  if (c >= '0' && c <= '9') {
+    return c - '0';
+  }
+
+  if (c >= 'A' && c <= 'F') {
+    return 10 + c - 'A';
+  }
+
+  if (c >= 'a' && c <= 'f') {
+    return 10 + c - 'a';
+  }
+  return -1;
+}
+
+template <typename T>
+struct FromHexFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE bool call(
+      out_type<Varchar>& result,
+      const arg_type<Varbinary>& input) {
+    const auto resultSize = (input.size() + 1) >> 1;
+    result.resize(resultSize);
+    const char* inputBuffer = input.data();
+    char* resultBuffer = result.data();
+
+    int32_t i = 0;
+    if ((input.size() & 0x01) != 0) {
+      if (inputBuffer[0] < 0) {
+        return false;
+      }
+      auto v = fromHex(inputBuffer[0]);
+      if (v == -1) {
+        return false;
+      }
+      // padding with '0' to align with Spark.
```

Comment:
nit: capitalize the first letter for comment.

Reply:
done, thanks!